nifi-commons/nifi-socket-utils/src/main/java/org/apache/nifi/io/socket/SocketUtils.java

```diff
@@ -188,6 +188,9 @@ public static void closeQuietly(final Socket socket) {
                 }
             } finally {
                 if (!socket.isClosed()) {
+                    if (socket.getChannel().isOpen()) {
+                        socket.getChannel().close();
+                    }
```

Comment:
Is this change necessary?  According to the `java.net.Socket` documentation, `getChannel()` can return null, so if this change is necessary, the result of `getChannel()` should be checked for `null`.